.github/workflows/build_and_test.yml

```diff
@@ -204,13 +243,26 @@ jobs:
       - name: Build Tutorials
         run: |
           mkdir -p build_examples
-          cmake -DPCAPPP_BUILD_TUTORIALS=ON -DCMAKE_OSX_ARCHITECTURES=${{ matrix.arch }} -S Examples -B build_examples
+          cmake -DPCAPPP_BUILD_TUTORIALS=ON -DPCAPPP_BUILD_COVERAGE=ON -DCMAKE_OSX_ARCHITECTURES=${{ matrix.arch }} -S Examples -B build_examples
           cmake --build build_examples -j
 
       - name: Test Tutorials
         if: ${{ matrix.arch == 'x86_64' }}
         run: cd build_examples/tutorials_bin && ./Tutorial-HelloWorld
 
+      - name: Create Cobertura Report
+        run: |
+          python3 -m pip install gcovr
+          gcovr -v -r . --gcov-ignore-parse-errors --exclude-throw-branches --filter Common --filter Pcap --filter Packet --xml -o coverage.xml
+
+      - name: Upload Coverage Results
+        uses: codecov/codecov-action@v3
+        with:
+          files: ./coverage.xml
+          flags: ${{ matrix.os-version }},unittest
+          fail_ci_if_error: true
```

Comment:
do we want the CI to fail on coverage upload errors? 🤔

Reply:
Both options are ok for me. It is up to you

Reply:
IMO it's should be set to `false` so if codecov is down our CI won't fail